control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -294,6 +310,48 @@ func (r *HostedControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 		}
 	}
 
+	// Reconcile etcd cluster status
+	{
+		etcdCluster := etcd.Cluster(hostedControlPlane.Namespace)
+		var err error
+		if err = r.Get(ctx, types.NamespacedName{Namespace: etcdCluster.Namespace, Name: etcdCluster.Name}, etcdCluster); err != nil && !apierrors.IsNotFound(err) {
+			return ctrl.Result{}, fmt.Errorf("failed to fetch etcd cluster %s/%s: %w", etcdCluster.Namespace, etcdCluster.Name, err)
+		}
+		if apierrors.IsNotFound(err) {
+			etcdCluster = nil
+		} else if !etcdCluster.DeletionTimestamp.IsZero() {
+			// Wait til etcd cluster is gone in case it's being deleted
+			return ctrl.Result{RequeueAfter: etcdDeleteCheckInterval}, nil
+		}
+		err = etcd.ReconcileEtcdClusterStatus(ctx, r.Client, hostedControlPlane, etcdCluster)
```

Comment:
Deletion happening of a side effect of status calc is surprising and inconsistent, seems like a small controller or another reconcile function later on?

Reply:
Yeah, this is kind of a special case, since most things we simply CreateOrUpdate. I can set status here and then based on that status delete the etcd cluster right before calling CreateOrUpdate.